Mathlib/LinearAlgebra/Contraction.lean

```diff
@@ -134,10 +158,13 @@ noncomputable def dualTensorHomEquivOfBasis : Module.Dual R M ⊗[R] N ≃ₗ[R]
         Fintype.sum_apply, Function.comp_apply, Basis.coe_dualBasis, coe_comp, compr₂_apply,
         tmul_smul, smul_tmul', ← sum_tmul, Basis.sum_dual_apply_smul_coord])
 
+theorem coe_dualTensorHomEquivOfBasis :
+    ⇑(dualTensorHomEquivOfBasis (N := N) b) = dualTensorHom R M N := rfl
+
 @[simp]
 theorem dualTensorHomEquivOfBasis_apply (x : Module.Dual R M ⊗[R] N) :
     dualTensorHomEquivOfBasis b x = dualTensorHom R M N x := by
-  ext; rfl
+  rfl
```

Comment:
I think for this to work with `dsimp` (in a recent or upcoming lean version) you also have to remove the `by`